### ISSUE TITLE
Error is not reported when TypeAliasType definition is cyclical

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#10475]
When using the PEP 695 `type` statement, it's possible for a type alias definition to reference the type alias symbol itself, allowing for cyclical or recursive types. However, the `TypeAliasType` constructor, which is used for backward compatibility, does not allow this. 

Pyright does not currently report an error in this situation.

```python
BadAlias21 = TypeAliasType("BadAlias21", list[BadAlias21]) # No type error, but runtime error
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.401.
